### PR TITLE
Latest page - full redesign

### PR DESCRIPTION
**Pull request by Joe-Brady**
**TO BE SQUASHED INTO ONE COMMIT ON MERGE**

[On staging](https://staging.adoptopenjdk.net/releases.html)

Entirely new design:

- **Clean appearance:** hides away the 'mess' into a tab-style platform selector system
- **Ease of use:** very clear UI, obvious what you should select.
- **Installer-ready:** if Windows `.exe` or macOS `.dmg` installers are added to the releases repo, they will appear automatically on this page on the website. Other platforms' installers will need their extension (e.g. `.deb`) adding to the global platforms array - I didn't add them this time because I wasn't sure what the extensions would be.
- **Future-proof:** anything else can be added into the platform information area - e.g. platform information, future download types, more alternative versions, etc. Also won't look strange if every platform has a slightly different set of downloads/information, because you only see one at a time.
- **Mobile responsive:** every aspect works on every screen size, and a prompt to 'Scroll horizontally' appears above the platform selector when the screen is not wide enough to display all platforms.
- **More prominent build name/number:** as per @karianna 's request

### CODE REVIEW

File: src/handlebars/releases.handlebars [hunk 1]
```diff
@@ -4,31 +4,48 @@
...
 
     <div id="latest-container" class="invisible">
 
+      <h2 style="margin-bottom: 0.6rem;">Build <strong><a href="" id="latest-build-name" target="_blank"></a></strong></h2>
+
       <table id="latest-metadata-table" cellspacing="0" cellpadding="0">
         <tr>
-          <td><strong>Name: </strong><a href="" id="latest-build-name" class="dark-link no-underline" target="_blank"></a></td>
+          <!--<td><a href="" id="latest-build-name" class="dark-link no-underline" target="_blank"></a></td>-->
           <td><strong>Date: </strong><span id="latest-date"></span></td>
```
Comment (karianna) [inline src/handlebars/releases.handlebars]: remove commented out code?

File: src/handlebars/releases.handlebars [hunk 2]
```diff
@@ -4,31 +4,48 @@
...
-          <!--<td><a href="" class="light-link" id="latest-changelog" target="_blank"><strong>Changelog</strong></a></td>-->
           <td><strong>Timestamp: </strong><span id="latest-timestamp"></span></td>
-          <!--<td><strong>Build number: </strong><span id="latest-buildnumber"></span></td>-->
+          <!--<td><a href="" class="light-link" id="latest-changelog" target="_blank"><strong>Changelog</strong></a></td>-->
           <!--<td><strong>Commit ref: </strong><a href="" class="light-link" id="latest-commitref" target="_blank"></a></td>-->
         </tr>
```
Comment (karianna) [inline src/handlebars/releases.handlebars]: remove commented out code?

File: src/js/0-global.js [hunk 3]
```diff
@@ -12,47 +12,53 @@ var platforms = [
     officialName: "Linux x86-64",
     searchableName: "X64_LINUX",
     logo: "linux.png",
-    fileExtension: ".tar.gz",
+    binaryExtension: ".tar.gz",
+    installerExtension: "no-installer-available",
     architecture: "64",
```
Comment (karianna) [inline src/js/0-global.js]: I like this idea - will be interesting when we produce rpms and .debs

File: src/js/nightly.js [hunk 4]
```diff
@@ -87,14 +87,14 @@ function buildNightlyHTML(releasesJson) {
...
+          var newNightlyContent = currentNightlyContent += ("<tr class='nightly-container'><td class='nightly-header'><div><strong><a href='"+thisGitLink+"' class='dark-link' target='_blank'>"+thisReleaseName+"</a></strong></div><div class='divider'> | </div><div class='nightly-release-date'>"+thisReleaseDate+"</div></td><td class='nightly-platform-block'>"+thisOfficialName+"</td><td class='nightly-downloads-block'><div><a class='dark-link' href='"+thisBinaryLink+"'>"+thisBinaryExtension+" ("+thisBinarySize+" MB)</a><div class='divider'> | </div><a href='"+thisChecksumLink+"' class='dark-link'>Checksum</a></div></td><td class='nightly-details'><!--<div><strong><a href='put-changelog-link-here' class='dark-link'>Changelog</a></strong></div> <div class='divider'> | </div>--><div><strong>Timestamp: </strong>"+thisTimestamp+"</div><!--<div class='divider'> | </div> <div><strong>Commit: </strong><a href='put-commit-ref-link-here' class='dark-link'>put-commit-ref-here</a></div>--></td></tr>");
```
Comment (karianna) [inline src/js/nightly.js]: Seems to be some commented out HTMl in this

File: src/js/releases.js [hunk 5]
```diff
@@ -30,9 +30,8 @@ function buildLatestHTML(releasesJson) {
   document.getElementById("latest-build-name").innerHTML = releasesJson.name;
   document.getElementById("latest-build-name").href = ("https://github.com/AdoptOpenJDK/openjdk-releases/releases/tag/" + releasesJson.name);
   document.getElementById("latest-date").innerHTML = moment(publishedAt).format('Do MMMM YYYY');
-  //document.getElementById("latest-changelog").href = releasesJson.name;
   document.getElementById("latest-timestamp").innerHTML = (publishedAt.slice(0, 4) + publishedAt.slice(8, 10) + publishedAt.slice(5, 7) + publishedAt.slice(11, 13) + publishedAt.slice(14, 16));
-  //document.getElementById("latest-buildnumber").innerHTML = releasesJson.id;
+  //document.getElementById("latest-changelog").href = releasesJson.name;
   //document.getElementById("latest-commitref").innerHTML = releasesJson.name;
   //document.getElementById("latest-commitref").href = releasesJson.name;
```
Comment (karianna) [inline src/js/releases.js]: commented out code?

File: src/js/releases.js [hunk 6]
```diff
@@ -51,33 +53,78 @@ function buildLatestHTML(releasesJson) {
     // firstly, check if the platform name is recognised...
     if(thisPlatform) {
 
-      // secondly, check if the file has the expected file extension for that platform...
+      // secondly, search the assetArray for an installer for this platform, and create an HTML block for it
+      var thisInstallerBlock = "";
+      var thisInstallerExtension = getInstallerExt(thisPlatform); // get the file extension associated with this platform
+
+      assetArray.forEach(function(eachAsset2) {
+        var nameOfFile2 = (eachAsset2.name);
```
Comment (karianna) [inline src/js/releases.js]: eachAsset2 is a code smell :-)  What's the difference between eachAsset and eachAsset2?  And exactly what sort of asset is this?

Comment (karianna) [inline src/js/releases.js]: Same goes for the other variables you've got here called xxx2

**Comment by karianna** [comment]
New design looks great!

**Comment by Joe-Brady** [comment]
Appears to also be some issues on FireFox and Safari, due to the `display: inline-grid` CSS value

**Comment by gdams** [comment]
linter failed: https://ci.adoptopenjdk.net/view/website/job/adoptopenjdk-website-staging/128/console

**Comment by Joe-Brady** [comment]
@karianna @gdams All fixed now, live on https://staging.adoptopenjdk.net/releases.html

File: src/js/0-global.js (hunk 3 above)
Comment (gdams) [inline src/js/0-global.js]: yup, mac and windows will be easy, linux should get more interesting

**Pull request merged**